pkg/controllers/management/auth/crtb_handler.go

```diff
@@ -3,21 +3,52 @@ package auth
 import (
 	"fmt"
 	"strings"
+	"time"
 
 	"github.com/hashicorp/go-multierror"
 	"github.com/pkg/errors"
 	"github.com/rancher/rancher/pkg/controllers/management/authprovisioningv2"
+	mgmtconv3 "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
 	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
 	typesrbacv1 "github.com/rancher/rancher/pkg/generated/norman/rbac.authorization.k8s.io/v1"
 	pkgrbac "github.com/rancher/rancher/pkg/rbac"
 	"github.com/rancher/rancher/pkg/user"
 	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/util/retry"
 )
 
+const (
+	SummaryInProgress = "InProgress"
+	SummaryCompleted  = "Completed"
+	SummaryError      = "Error"
+)
+
+// Condition reason types
+const (
```

Comment:
Can we expand on these and explain under what circumstances each is used?

Reply:
Each const is pretty much named after the circumstance using it.
Note, that despite using public names these const's are only used in the file defining them.
I.e. it is trivial to locate the using places by grepping. Each should have about 1 to 2 hits at most.

Reply:
Sure, but customers will see these, and it likely won't be quite as obvious to them?

Reply:
You mean we should document them in our official public documentation for Rancher ?
Because any documentation in code, as I thought that you asked for, would not help users either.

Reply:
Just as an example, Pods have the following documentation for their conditions: https://github.com/kubernetes/kubernetes/blob/7f6f68fdabc4df88cfea2dcf9a19b2b830f1e647/pkg/apis/core/types.go#L2538

I'm not sure all of them are particularly valuable, `PodScheduled represents status of the scheduling process for this pod` is pretty obvious even without the comment. I'm kind of 50/50 on the idea of documenting each failure type and how much value it really adds.

Reply:
Addressed with commit